src/foam/nanos/dig/DIG.js

```diff
@@ -53,6 +53,16 @@ foam.CLASS({
     },
     'cmd',
     'format',
+    {
+      class: 'StringArray',
+      name: 'fields',
+      expression: function(daoKey) {
+         var dao = this.__context__[daoKey];
+
+         if ( dao != null )
```

Comment:
You do not need this if statement now that you check if dao is null in the return statement.